src/hostcalls.rs

```diff
@@ -1576,6 +1584,43 @@ fn get_hostfunc(
         }
 
         /* ---------------------------------- System ---------------------------------- */
+        "clock_time_get" => Some(Func::wrap(
+            store,
+            |mut _caller: Caller<'_, ()>, _clock_id: i32, _precision: i64, _time: i32| -> i32 {
+                Status::Ok as i32
+            },
+        )),
+
+        "random_get" => Some(Func::wrap(
+            store,
+            |mut _caller: Caller<'_, ()>, _buf: i32, _buf_len: i32| -> i32 { Status::Ok as i32 },
+        )),
+
+        "fd_write" => Some(Func::wrap(
+            store,
+            |mut _caller: Caller<'_, ()>,
+             _param1: i32,
+             _param2: i32,
+             _param3: i32,
+             _param4: i32|
+             -> i32 { Status::Ok as i32 },
+        )),
+
+        "environ_get" => Some(Func::wrap(
```

Comment:
was that autogenerated code? If so, how did you generate it?

Reply:
Not autogenerated. 

You have to run the abi_binary_interface command in order to see what the warm module exports and imports. These were the missing imports with wasi.


However, in the meantime I found a nice example of a WASM module that is using the existing test-framework and they updated the code to work with ABI 0.2.0, but not using the wash interface.

AFAICT while there has been a relative amount of movement to wasi, we can use the main branch of test-framework and use the `wasm32-unknown-unknown` toolchain for the mean time. That way we don't need to fork or maintain the test-framework. 

That being said, I am going to land this and we can leave this fork alive in case Katanemo's priority lines up to spend more time here.

For now I will spend time writing actual tests for the intelligent-prompt-gateway.

Reply:
NVM, upon further investigation, even though https://github.com/proxy-wasm/test-framework/pull/16 did a lot of work to catch up to 0.2.0 we still need the fork because test-framework is not fully 0.2.0 compatible. For instance, all the metrics ABI is outdated. So I will continue working on this fork while writing [the initial tests in the gateway code](https://github.com/katanemo/intelligent-prompt-gateway/pull/20).